### PR TITLE
GEKO-7

### PR DESCRIPTION
**Pull request by chepix10**
- Moved 400 responses to dynamic deserialization
- All Bad Request responses now have validation_details on their body if it's returned by the API call
- Added assertions for 400 responses in all tests
- Bumped version to 0.2.7-rc-3

### CODE REVIEW

File: test/Api/GetAccountBalanceTest.php [hunk 1]
```diff
@@ -74,6 +66,9 @@ public function testGetAccountBalance400()
         $resp = explode("||", $data);
         $response = self::$api->getAccountBalance(0, $resp[1], 'default');
         $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(false, $response->getData()->success);
+        $this->assertStringContainsString('Bad request', $response->getData()->message);
+        $this->assertTrue($response->getData()->validation_details != null);
     }
 
 }
```
Comment (ArcGate-Satyendra) [inline test/Api/GetAccountBalanceTest.php:74]: Add newline at EOF

**Comment by ArcGate-Satyendra** [comment]
**Verified earlier response on API:**
`    [data:Silamoney\Client\Api\ApiResponse:private] => Silamoney\Client\Domain\BaseResponse Object
        (
            [reference:Silamoney\Client\Domain\BaseResponse:private] => 
            [message:Silamoney\Client\Domain\BaseResponse:private] => Bad request.
            [status:Silamoney\Client\Domain\BaseResponse:private] => FAILURE
        )`

**After Changes:**

`   [data:Silamoney\Client\Api\ApiResponse:private] => stdClass Object
        (
            [validation_details] => stdClass Object
                (
                    [crypto_entry] => stdClass Object
                        (
                            [crypto_address] => This field may not be blank.
                        )
                )
            [message] => Bad request.
            [status] => FAILURE
        )
`

**Comment by ArcGate-Satyendra** [comment]
I'm going to approve this PR, but please add a newline at EOF prior to merging.

File: test/Api/GetAccountBalanceTest.php (hunk 1 above)
Comment (chepix10) [reply test/Api/GetAccountBalanceTest.php]: Done. Thanks for catching that one.

**Comment by chepix10** [comment]
@ArcGate-Satyendra I've also bumped the version on the README

**Pull request merged**